### PR TITLE
Fix #16370 - Added view/DDL definition rule

### LINKED ISSUE #16370: Adding view/DDL definition rule.
Hi OMD Team! 
We at the organization believe that view definition code is sensitive information. We tried to deny its viewing at the privilege level, but this only works with the Deny ViewAll rule for Table.

I see the following solution:
Add a rule to view the view definition and DDL definition in the access settings.

Alternatively, you can add a switch to Metadata Ingestion. Like a "Include DDL" option.

### PR DESCRIPTION
**Pull request by SumanMaharana**
### Describe your changes:

Fixes #16370 

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py [hunk 1]
```diff
@@ -615,26 +615,27 @@ def get_schema_definition(
         Get the DDL statement or View Definition for a table
         """
         try:
-            if table_type == TableType.View:
-                view_definition = inspector.get_view_definition(
-                    fqn._build(self.context.get().database, schema_name, table_name)
+            if self.source_config.includeDDL:
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/bigquery/metadata.py:618]: We should do that check where we call `get_schema_definition` so `get_schema_definition` actually get only the schema definition

File: ingestion/src/metadata/ingestion/source/database/clickhouse/metadata.py [hunk 2]
```diff
@@ -153,19 +153,20 @@ def get_schema_definition(
         Get the DDL statement or View Definition for a table
         """
         try:
-            if table_type in {TableType.View, TableType.MaterializedView}:
-                definition_fn = inspector.get_view_definition
-                schema_definition = definition_fn(table_name, schema_name)
-            else:
-                schema_definition = inspector.get_table_ddl(
-                    self.connection, table_name, schema_name
+            if self.source_config.includeDDL:
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/clickhouse/metadata.py:156]: same question here regarding the check

File: ingestion/src/metadata/ingestion/source/database/clickhouse/metadata.py [hunk 3]
```diff
@@ -153,19 +153,20 @@ def get_schema_definition(
...
+                if table_type in {TableType.View, TableType.MaterializedView}:
```
Comment (TeddyCr) [inline ingestion/src/metadata/ingestion/source/database/clickhouse/metadata.py:157]: let's keep the code style constant if we do `if;return;return` let's do this everywhere and not switch between `if;return;return` and `if;else;return`

Comment (SumanMaharana) [reply ingestion/src/metadata/ingestion/source/database/clickhouse/metadata.py:157]: will change the code style into `if;return;return`

**Review by TeddyCr — APPROVED** [review]

**Pull request merged**